Firestore/core/test/unit/api/aggregate_query_test.cc

```diff
@@ -0,0 +1,160 @@
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+
+using model::AggregateAlias;
+using util::Status;
+
+namespace api {
+
+class MockAggregateQuery : public AggregateQuery {
+ public:
+  StatusOr<ObjectValue> mockResult;
+
+  ~MockAggregateQuery() override = default;
+
+  MockAggregateQuery(Query query,
+                     std::vector<AggregateField>&& aggregates,
+                     StatusOr<ObjectValue>&& mockResult)
+      : AggregateQuery(query, std::move(aggregates)),
+        mockResult(std::move(mockResult)){};
+
+  void GetAggregate(AggregateQueryCallback&& callback) override {
+    callback(mockResult);
+  }
+};
+
+class AggregateQueryTest {
+ public:
+  static const Query& GetQuery(const AggregateQuery& aggregateQuery) {
+    return aggregateQuery.query_;
+  }
+
+  static const std::vector<AggregateField>& GetAggregates(
+      const AggregateQuery& aggregateQuery) {
+    return aggregateQuery.aggregates_;
+  }
+};
+
+namespace {
+
+// Assert that the Get method calls GetAggregate method
+// and that the result from GetAggregate is processed
+// appropriately
+TEST(AggregateQuery, GetCallsGetAggregateOk) {
+  // Params for Get: Query and AggregateField vector
+  Query query;
+  AggregateField countAggregateField(AggregateField::OpKind::Count,
+                                     AggregateAlias("count"));
+  std::vector<AggregateField> aggregates{countAggregateField};
+
+  // Mock aggregate field result
+  google_firestore_v1_AggregationResult_AggregateFieldsEntry
+      mockAggregatesField[1];
+  mockAggregatesField[0].key =
+      nanopb::ByteString(absl::string_view("count")).release();
+  mockAggregatesField[0].value.which_value_type =
+      google_firestore_v1_Value_integer_value_tag;
+  mockAggregatesField[0].value.integer_value = 10;
+
+  // Mock ObjectValue result
+  ObjectValue mockObjectValue =
+      ObjectValue::FromAggregateFieldsEntry(mockAggregatesField, 1);
+
+  // Set mocked result on our MockAggregateQuery object
+  MockAggregateQuery aggregateQuery(query, std::move(aggregates),
+                                    StatusOr<ObjectValue>(mockObjectValue));
+
+  // Call the Get method
+  size_t callbackCount = 0;
+  aggregateQuery.Get([&callbackCount](const StatusOr<int64_t>& result) mutable {
+    callbackCount++;
+    EXPECT_EQ(result.ok(), true);
+    EXPECT_EQ(result.ValueOrDie(), 10);
+  });
+
+  // Assert
+  EXPECT_EQ(callbackCount, 1);
+}
+
+// Assert that the Get method calls GetAggregate method
+// and that an error result from GetAggregate is processed
+// appropriately
+TEST(AggregateQuery, GetCallsGetAggregateError) {
+  // Params for Get: Query and AggregateField vector
+  Query query;
+  AggregateField countAggregateField(AggregateField::OpKind::Count,
+                                     AggregateAlias("count"));
+  std::vector<AggregateField> aggregates{countAggregateField};
+
+  // Mock error status
+  Status mockError = Status(Error::kErrorInternal, "foo");
```

Comment:
Please rename `mockError` to have a different prefix. The term "mock" is specifically referring to a mock object created by gmock. In this case, it's not a mock object, it's a real object. Consider calling it "sample_error" or "simulated_error" or even just "error".

Reply:
Changed this even though I disagree. Mock is a concept that extends outside of gmock. In this case, the object is not mocking out an interface with fake behavior, the object does mock a status result from a lower layer.

Reply:
noted